src/Pim/Bundle/CatalogBundle/Elasticsearch/Filter/Attribute/MetricFilter.php

```diff
@@ -0,0 +1,245 @@
+                $clause = [
+                    'range' => [
+                        $attributePath => ['lte' => $value]
+                    ]
+                ];
+                $this->searchQueryBuilder->addFilter($clause);
+                break;
+
+            case Operators::EQUALS:
+                $clause = [
+                    'term' => [
+                        $attributePath => $value
+                    ]
+                ];
+                $this->searchQueryBuilder->addFilter($clause);
+                break;
+
+            case Operators::NOT_EQUAL:
+                $mustNotClause = [
+                    'term' => [
+                        $attributePath => $value,
+                    ],
+                ];
+                $filterClause = [
+                    'exists' => [
+                        'field' => $attributePath,
+                    ],
+                ];
+                $this->searchQueryBuilder->addMustNot($mustNotClause);
+                $this->searchQueryBuilder->addFilter($filterClause);
+                break;
+
+            case Operators::GREATER_OR_EQUAL_THAN:
+                $clause = [
+                    'range' => [
+                        $attributePath => ['gte' => $value]
+                    ]
+                ];
+                $this->searchQueryBuilder->addFilter($clause);
+                break;
+
+            case Operators::GREATER_THAN:
+                $clause = [
+                    'range' => [
+                        $attributePath => ['gt' => $value]
+                    ]
+                ];
+                $this->searchQueryBuilder->addFilter($clause);
+                break;
+
+            case Operators::IS_EMPTY:
+                $clause = [
+                    'exists' => [
+                        'field' => $attributePath
+                    ]
+                ];
+                $this->searchQueryBuilder->addMustNot($clause);
+                break;
+
+            case Operators::IS_NOT_EMPTY:
+                $clause = [
+                    'exists' => [
+                        'field' => $attributePath
+                    ]
+                ];
+                $this->searchQueryBuilder->addFilter($clause);
+                break;
+
+            default:
+                throw InvalidOperatorException::notSupported($operator, static::class);
+        }
+
+        return $this;
+    }
+
+    /**
+     * Check if the given value is valid
+     *
+     * @param AttributeInterface $attribute
+     * @param mixed              $data
+     *
+     * @throws InvalidPropertyTypeException
+     * @throws InvalidPropertyException
+     */
+    protected function checkValue(AttributeInterface $attribute, $data)
+    {
+        if (!is_array($data)) {
+            throw InvalidPropertyTypeException::arrayExpected($attribute->getCode(), static::class, $data);
+        }
+
+        if (!array_key_exists('amount', $data)) {
```

Comment:
Just an idea, not to do in this PR, but there is really something to mutualize here: we perform the same checks in the product value factories, in the copiers and the adders. And it is the same for all attribute types.